src/moonstone-samples/lib/DataGridListSample.js

```diff
@@ -110,12 +151,70 @@ module.exports = kind({
 		this.modelIndex = idx;
 		return records;
 	},
+	generateDataGridList: function (itemType) {
+		if (this.$.gridList) {
+			this.$.gridList.destroy();
+		}
+
+		var props = utils.clone(this.dataListDefaults),
```

Comment:
Hi @DavidUM, this is not performing a deep-clone, so when we modify a nested property of `props` i.e. in the case of "HorizontalGridListItem" setting the bindings array, we are changing the bindings of "GridSampleItem". Switching from "HorizontalGridListItem" to "GridSampleItem" results in the `source` binding being effectively removed, so no image shows up. I think we'll have to rework how we're generating the item properties.